yaksh/views.py

```diff
@@ -1025,3 +1036,92 @@ def edit_profile(request):
         context['form'] = form
         return my_render_to_response('yaksh/editprofile.html', context,
                                     context_instance=ci)
+
+
+@login_required
+def search_teacher(request, course_id):
+    """ search teachers for the course """
+    user = request.user
+    ci = RequestContext(request)
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+
+    context = {}
+    course = get_object_or_404(Course, creator=user, pk=course_id)
+    context['course'] = course
+
+    if request.method == 'POST':
+        u_name = request.POST.get('uname')
+        if len(u_name) == 0:
+            return my_render_to_response('yaksh/addteacher.html', context,
+                                        context_instance=ci)
+        else:
+            teachers = User.objects.filter(Q(username__icontains=u_name)|
+                Q(first_name__icontains=u_name)|Q(last_name__icontains=u_name)|
+                Q(email__icontains=u_name)).exclude(Q(id=user.id)|Q(is_superuser=1))
+            context['success'] = True
+            context['teachers'] = teachers
+            return my_render_to_response('yaksh/addteacher.html', context,
+                                        context_instance=ci)
+    else:
+        return my_render_to_response('yaksh/addteacher.html', context,
+                                    context_instance=ci)
+
+
+@login_required
+def add_teacher(request, course_id):
+    """ add teachers to the course """
+
+    user = request.user
+    ci = RequestContext(request)
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+
+    context = {}
+    course = get_object_or_404(Course, creator=user, pk=course_id)
+    context['course'] = course
+
+    if request.method == 'POST':
+        teacher_ids = request.POST.getlist('check')
+        teachers = User.objects.filter(id__in=teacher_ids)
+        teachers_added, teachers_rejected = course.add_teachers(*teachers)
+        context['status'] = True
+        context['teachers_added'] = teachers_added
+        context['teachers_rejected'] = teachers_rejected
+        return my_render_to_response('yaksh/addteacher.html', context,
+                                    context_instance=ci)
+    else:
+        return my_render_to_response('yaksh/addteacher.html', context,
+                                    context_instance=ci)
+
+
+@login_required
+def view_courses(request):
```

Comment:
I think instead of view_courses it can be allotted_courses.
Also, in template you have viewcourses, which can be allotted_courses.